### DISCUSSION TITLE
PRESENTACIÓN DE PROYECTO FINAL VIERNES 14 de JUNIO

### DISCUSSION THREAD

**Discussion by harpomaxx** [harpomaxx/algoritmos2#79] [Anuncios]
**Respecto al proyecto final:**

Tienen hasta el Jueves a las 23:59 para entregar el proyecto. Es decir para hacer el push del último commit. Cualquier commit después de esa hora no será considerado. Recuerden que necesitamos tiempo para mirar si sus implementaciones funcionan. 

Los proyectos tienen que tener funcionando:
a. La carga de los PDF
b. La búsqueda de una palabra clave (puede ser una cadena)
c. Algún mecanismo propuesto para realizar búsqueda semántica. (traer en el ranking documentos relacionados)

Si alguna de las 3 no funciona, el proyecto se considera desaprobado.

A las 16:00hs del viernes 14 de junio, vamos a comenzar con las presentaciones. Las mismas van a ser de manera virtual. Igual que la  última vez (el mismo jitsi de siempre). Para ello van a tener que armar **una presentación de no más de 10 minutos de duración**

**En la presentación les vamos a pedir:**

a. Problemas encontrados durante el desarrollo. Ej. cómo manejar el path, como crear la db de documento, cómo mantener la persistencia, como hacer la búsqueda de palabras, entre otros.

b. Soluciones a los diferentes problemas. Básicamente que enfoque algorítmico aplicaron o que estructura usaron.

c. Una estimación de la complejidad espacial o temporal (en caso de ser relevante) para las funciones

d. Posibles mejoras de su solución (no implementadas por falta de tiempo)

e. Además de la solución propuesta, vamos a evaluar el tiempo de su exposición, la calidad de la presentación y que todos demuestren participación y conocimiento sobre lo expuesto.

### IMPORTANTE: NO QUEREMOS VER DETALLES DE CÓDIGO EN LA PRESENTACIÓN, DEBERÁN INCLUIR DIAGRAMAS QUE EXPLIQUEN SU SOLUCIÓN

**Comment by LuciiAl** [comment]
Hola profe, perdon la hora, pero estamos viendo con mis compañeros de grupo que no entendemos muy bien que deberiamos hacer en el inciso b
Debemos recibir una palabra clave o cadena y devolver en que documento se encuentra? Y si se encuentra en mas de uno devolver todos? 
No se nos ocurre que otra cosa puede ser y cual sería la diferencia del inciso c.

**Comment by harpomaxx** [comment]
en b) me refiero a que la aplicacion debe poder recibir una cadena de texto, no solo una palabra. en c) es la busqueda puede ser precisa o semantica. (claro que alguna forma de busqueda semantica deben implementar.. por que es justamente lo que vamos a evaluar tambien.). Se entiende?

**Reply by LuciiAl** [reply]
Ah perfecto, gracias profe!
